avocado/core/parser.py

```diff
@@ -46,6 +51,32 @@ def _get_option_tuples(self, option_string):
         return []
 
 
+class StdoutAlreadyClaimedError(Exception):
+
+    """
+    Signals that other player already claimed exclusive rights on STDOUT
+    """
+
+
+class FileOrStdoutAction(argparse.Action):
```

Comment:
I think you got my comment wrong. I like the `Action` and it should be implemented the way it is. The difference was that instead of `stdout_claimed` I'd simply create `claim_stdout()` method, which would be used inside the `Action`.

The function `claim_stdout(option)` would check if `self._claim_stdout` is set and raise exception. Otherwise it'd store the claimer (option) so the next claim could fail with a nice exception.

The difference is that `stdout_claimed` is a variable and if one needs a different store action (eg. output only to stdout, or few other options) they'd have to handle the `stdout_claimed` manually. On the other hand `claim_stdout` would be easy to use for other custom actions.

Although I haven't looked at the code and the `parser` is still being constructed at the time it's used, so I'm not sure it's possible to write it the way I described it. So feel free to use other ways. My main concern here was the use of global variable, which in my eyes looked non-necessary.